File: controller_math.py
class RoboticArm:
    def __init__(self, initial_x=0.0, initial_y=0.0):
        self.x = initial_x
        self.y = initial_y

    def update_position(self, joystick_left, joystick_right):
        # Scale joystick input to a maximum of 2 inches
        x_offset = round(joystick_left * 2.0, 1)
        y_offset = round(joystick_right * 2.0, 1)

        # Update current position
        self.x += x_offset
        self.y += y_offset

        print(f"New Position -> X: {self.x} inches, Y: {self.y} inches")

    def get_position(self):
        return self.x, self.y

if __name__ == "__main__":
    arm = RoboticArm()  # Start at the origin

    # Simulate joystick inputs
    left_joystick = -0.5  # Up 100%
    right_joystick = 0.5  # Down 100%

    # Update the robotic arm position based on the joystick inputs
    arm.update_position(left_joystick, right_joystick)

    # Get and display the updated position
    current_x, current_y = arm.get_position()
    print(f"Updated Position -> X: {current_x}, Y: {current_y}")

File: old/controller_math.cpp
#include <iostream>
#include <cmath>

class RoboticArm {
private:
    float x, y; // Current coordinates of the robotic arm's end effector

public:
    RoboticArm(float initialX = 0.0, float initialY = 0.0) : x(initialX), y(initialY) {}

    // Function to update the position based on joystick input
    void updatePosition(float joystickLeft, float joystickRight) {
        // Scale joystick input to a maximum of 2 inches
        float xOffset = round(joystickLeft * 2.0 * 10) / 10.0;
        float yOffset = round(joystickRight * 2.0 * 10) / 10.0;

        // Update current position
        x += xOffset;
        y += yOffset;

        std::cout << "New Position -> X: " << x << " inches, Y: " << y << " inches\n";
    }

    // Function to get the current position
    void getPosition(float& currentX, float& currentY) {
        currentX = x;
        currentY = y;
    }
};

int main() {
    RoboticArm arm(0.0, 0.0); // Starting at origin

    // Simulate joystick inputs
    float leftJoystick = -1.0; // Up 100%
    float rightJoystick = 1.0; // Down 100%

    // Update the robotic arm position based on the joystick inputs
    arm.updatePosition(leftJoystick, rightJoystick);

    // Get and display the updated position
    float currentX, currentY;
    arm.getPosition(currentX, currentY);
    std::cout << "Updated Position -> X: " << currentX << ", Y: " << currentY << std::endl;

    return 0;
}
